addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -162,21 +167,30 @@ def raise_insufficiency_error(moves):
                 ("picking_type_id", "=", picking_type.id),
                 ("state", "=", "confirmed"),
             ]
-            limit = 1
+            limit = batch_size if reserve_all else to_reserve
             processed = Picking.browse()
+
             unsatisfied_state = lambda p: p.state not in ("assigned", "cancel", "done")
 
-            while reserve_all or to_reserve > 0:
-                if self:
-                    pickings = self.filtered(lambda p: p.picking_type_id == picking_type)
-                    # Removed processed pickings from self
-                    pickings -= processed
-                else:
-                    domain = base_domain[:]
-                    if processed:
-                        domain.append(("id", "not in", processed.ids))
-                    pickings = Picking.search(domain, limit=limit)
+            def generate_pickings():
+                """Abstract iteration over fetched pickings."""
+                while True:
+                    if self:
+                        # Removed processed pickings from self
+                        yield self.filtered(lambda p: p.picking_type_id == picking_type) - processed
+                    else:
+                        domain = base_domain[:]
+                        if processed:
+                            domain.append(("id", "not in", processed.ids))
+                        pickings = Picking.search(domain, limit=batch_size)
```

Comment:
nit: This should be `limit=limit` I think, but they look interchangeable

Reply:
not sure how this ended up showing in the diff, this looks to be before i made any changes :confused: `limit` param should be `batch_size` as there is no `limit` variable defined any more

Reply:
I thought i did that all in one commit :thinking: i'll go back and comb over them to make sure no changes got accidentally changes in other commits

Reply:
Ah, it was one of the ports from 11.0 improvements that changed it. Completely forgot about that!